Common++/header/IpAddresses.h

```diff
@@ -0,0 +1,551 @@
+		IPAddress(const IPv6Address& addr) : m_Type(IPv6AddressType), m_IPv6(addr) {}
+
+		/**
+		 * Overload of an assignment operator.
+		 * @param[in] addr A const reference to instance of IPv4Address
+		 */
+		inline IPAddress& operator=(const IPv4Address& addr);
+
+		/**
+		 * Overload of an assignment operator.
+		 * @param[in] addr A const reference to instance of IPv6Address
+		 */
+		inline IPAddress& operator=(const IPv6Address& addr);
+
+		/**
+		 * Gets the address type: IPv4 or IPv6
+		 * @return The address type
+		 */
+		AddressType getType() const { return static_cast<AddressType>(m_Type); }
+
+		/**
+		 * Returns a std::string representation of the address
+		 * @return A string representation of the address
+		 */
+		std::string toString() const { return (getType() == IPv4AddressType) ? m_IPv4.toString() : m_IPv6.toString();	}
+
+		/**
+		 * Determine whether the address is unspecified
+		 */
+		bool isUnspecified() const { return (getType() == IPv4AddressType) ? m_IPv4.isUnspecified() : m_IPv6.isUnspecified(); }
+
+		/**
+		 * Determine whether the object contains an IP version 4 address
+		 */
+		bool isIPv4() const { return getType() == IPv4AddressType; }
+
+		/**
+		 * Determine whether the object contains an IP version 6 address
+		 */
+		bool isIPv6() const { return getType() == IPv6AddressType; }
+
+		/**
+		 * Get a reference to IPv4 address instance
+		 * @return The const reference to IPv4Address instance
+		 */
+		const IPv4Address& getIPv4() const { return m_IPv4; }
+
+		/**
+		 * Get a reference to IPv6 address instance
+		 * @return The const reference to IPv6Address instance
+		 */
+		const IPv6Address& getIPv6() const { return m_IPv6; }
+
+		/**
+		 * Overload of the equal-to operator
+		 */
+		inline bool operator==(const IPAddress& rhs) const;
+
+		/**
+		 * Overload of the not-equal-to operator
+		 */
+		bool operator!=(const IPAddress& rhs) const { return !(*this == rhs); }
+
+
+		// Following methods are placed for backward compatibility and will be deleted in the future
+
+		/**
+		 * Clone the object
+		 * Notice this method is deprecated and will be deleted in the future
+		 * @return A newly allocated instance which is a clone of the current instance
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is now unnecessary and added for backward compatibility. It will be deleted in the future")]]
+		#endif
+		IPAddress *clone() const { return new IPAddress(*this); }
+
+		/**
+		 * Compare between this IP address and another IP address
+		 * Notice this method is deprecated and will be deleted in the future. Prefer an equal-to operator
+		 * @return True if addresses match or false otherwise
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is now unnecessary and added for backward compatibility. It will be deleted in the future. Prefer an equal-to operator")]]
+		#endif
+		bool equals(const IPAddress* other) const { return *this == *other; }
```

Comment:
if this method is not used anywhere I think we can remove it

Reply:
Removed